### PR TITLE
Add label-check action to pull requests

### PR DESCRIPTION
**Pull request by helto4real**
## Proposed change

This makes the builds fail if not PR are labeled correctly. Labels are used when doing release to make automatic release notes. This will make sure this is not forgotten before merging a PR.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code compiles without warnings (code quality chek)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/net-daemon/docs

### CODE REVIEW

File: .github/workflows/check_pr_label.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: 🏷️ Check labels
+on:
+  pull_request_target:
```
Comment (ludeeus) [inline .github/workflows/check_pr_label.yml:3]:
```suggestion
  pull_request:
```

File: .github/workflows/check_pr_label.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    types: [opened, labeled, unlabeled, synchronize]
```
Comment (ludeeus) [inline .github/workflows/check_pr_label.yml:4]:
```suggestion
    types: [opened, labeled, unlabeled, synchronize, reopened]
```

File: .github/workflows/check_pr_label.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    check_labels:
+        name: 🏷️ Check valid labels
+        runs-on: ubuntu-latest
+        steps:
+            - name: Check valid labels
+              uses: jesusvasquez333/verify-pr-label-action@v1.4.0
+              with:
+                github-token: '${{ secrets.GITHUB_TOKEN }}'
+                valid-labels: 'pr:bugfix, pr: enhancement, pr: dependency-update, pr: new feature'
+                pull-request-number: '${{ github.event.pull_request.number }}'
```
Comment (ludeeus) [inline .github/workflows/check_pr_label.yml:15]:
```suggestion
```

File: .github/workflows/check_pr_label.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+jobs:
+    check_labels:
+        name: 🏷️ Check valid labels
+        runs-on: ubuntu-latest
+        steps:
+            - name: Check valid labels
+              uses: jesusvasquez333/verify-pr-label-action@v1.4.0
+              with:
+                github-token: '${{ secrets.GITHUB_TOKEN }}'
+                valid-labels: 'pr:bugfix, pr: enhancement, pr: dependency-update, pr: new feature'
```
Comment (ludeeus) [inline .github/workflows/check_pr_label.yml:14]:
```suggestion
                valid-labels: >-
                  'pr:bugfix', 'pr: enhancement', 'pr: dependency-update', 'pr: new feature'
```

**Review by ludeeus — CHANGES_REQUESTED** [review]

File: .github/workflows/check_pr_label.yml (hunk 3 above)
Comment (ludeeus) [inline .github/workflows/check_pr_label.yml:15]:
```suggestion
                disable-reviews: true
```

File: .github/workflows/check_pr_label.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    types: [opened, labeled, unlabeled, synchronize]
+jobs:
+    check_labels:
+        name: 🏷️ Check valid labels
+        runs-on: ubuntu-latest
+        steps:
+            - name: Check valid labels
+              uses: jesusvasquez333/verify-pr-label-action@v1.4.0
+              with:
+                github-token: '${{ secrets.GITHUB_TOKEN }}'
```
Comment (ludeeus) [inline .github/workflows/check_pr_label.yml:13]:
```suggestion
                github-token: ${{ secrets.GITHUB_TOKEN }}
```

File: .github/workflows/check_pr_label.yml (hunk 4 above)
Comment (BeeHiveJava) [reply .github/workflows/check_pr_label.yml:14]: Also there needs to be a space added for the bugfix label like `pr: bugfix`.

Comment (helto4real) [reply .github/workflows/check_pr_label.yml]: that will not work

Comment (helto4real) [reply .github/workflows/check_pr_label.yml]: it does not work to have quotes like this it is a single line not a list

**Review by ludeeus — APPROVED** [review]

**Pull request merged**